builder/builder.go

```diff
@@ -310,28 +306,19 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 
 					// For push.
 					if shouldPush {
-						refKey := fmt.Sprintf("image-%d", imageIndex)
-						refPrefix := fmt.Sprintf("ref/%s", refKey)
-						imageIndex++
-
-						res.AddMeta(fmt.Sprintf("%s/image.name", refPrefix), []byte(saveImage.DockerTag))
-						res.AddMeta(fmt.Sprintf("%s/platform", refPrefix), []byte(platformStr))
-						res.AddMeta(fmt.Sprintf("%s/export-image-push", refPrefix), []byte("true"))
-						if saveImage.InsecurePush {
-							res.AddMeta(fmt.Sprintf("%s/insecure-push", refPrefix), []byte("true"))
+						_, err = gwCrafter.AddPushImageEntry(ref, imageIndex, saveImage.DockerTag, shouldPush, saveImage.InsecurePush, saveImage.Image, []byte(platformStr))
```

Comment:
The image index could be maintained in the crafter perhaps? This should simplify some things potentially.

... But then it would have to return the index as an ID and we'd have to keep track of it separately outside of the crafetr. Maybe it's ok the way it is then. (Feel free to ignore this comment)

Reply:
Ultimately that's the direction I want to head in as this code gets moved into the wait_block.go code that's called by popWaitBlock. However that should be left to a follow up PR (to limit the size of this one).